### PR TITLE
ARROW-5776: [Gandiva][Crossbow] Use commit id instead of fetch head.

### PR DESCRIPTION
**Pull request by praveenbingo**
- It is important to use the same commit to build the various gandiva artifacts
  on OsX/Linux etc, so that they are consistent.
- We use the commit id to pin all of the builds to the same commit on the tree.

### CODE REVIEW

**Comment by praveenbingo** [comment]
@ursabot crossbow package -g gandiva

**Comment by praveenbingo** [comment]
@kszucs The ursabot failure looks like a random job termination. Please take a look if this looks ok to you.

**Comment by praveenbingo** [comment]
@ursabot crossbow package -g gandiva

**Comment by kszucs** [comment]
@praveenbingo in order to crossbow work properly with ursabot We need to use FETCH_HEAD. Ursabot builds the magic refs provided by github: `refs/pull/<PR-id>/merge`, see an example [here](https://github.com/ursa-labs/crossbow/blob/ursabot-94-azure-conda-linux-gcc-py27/azure-pipelines.yml#L24). This magic reference doesn't exist until you push to the github remote.

Could You give me a little more context why it is not working for you? I'm sure We can find out a solution for both use cases.

**Comment by praveenbingo** [comment]
@kszucs 
We use crossbow to create Gandiva artifacts for both mac and linux and then we have an internal dremio job that collects those artifacts and packages it into a single maven fat jar.

We need all of the builds - Gandiva Mac, Gandiva Linux and the internal jar packager to work off the same versions in the tree to avoid inconsistencies. So we need this commit id in the cross bow file as an anchor to ensure that the three build jobs work using the same code.

Please let me know if this helps.

**Comment by wesm** [comment]
It seems like this synchronization issue is something that the script that runs the multiple Crossbow builds should take care of. All of the branches that you push to your Crossbow queue should be based off the same commit id

**Comment by praveenbingo** [comment]
@wesm Yes I agree. And it was achieved earlier using the commit id as part of the trigger to the crossbow branches.
Is there a replacement for the same (some flag to pass perhaps?)

**Comment by praveenbingo** [comment]
@kszucs Do you think we can push this in until we have an alternative?

**Comment by kszucs** [comment]
We should add a flag to crossbow whether to use fetch head or the commit sha. 
I wouldn't merge it until we're done with the release, but with a flag we could keep both requirements working.

**Comment by praveenbingo** [comment]
@kszucs Sounds good, i will unblock by releasing temporarily from a private fork. Will get to the requested change in a couple days. Thanks!

**Comment by praveenbingo** [comment]
@ursabot crossbow package -g gandiva

**Comment by praveenbingo** [comment]
@kszucs Can you please see if the current change is ok?

**Comment by praveenbingo** [comment]
sample build where commit id is used https://travis-ci.org/praveenbingo/arrow-build/builds/563952605

**Comment by pravindra** [comment]
@kszucs can you please review this change ?

**Comment by kszucs** [comment]
@pravindra where is `$CROSSBOW_USE_COMMIT_ID` set?

**Comment by praveenbingo** [comment]
@kszucs it is set in Travis env variables

**Comment by pravindra** [comment]
@kszucs this travis job picked up the env variable (CROSSBOW_USE_COMMIT_ID) and ran successfully. 

Please let's know if you want some change in the fix. Our (dremio) repo has been out-of-sync with arrow master for a while, and is blocked on this change.

**Comment by pravindra** [comment]
@kszucs  @wesm - i'll go ahead and merge this change tomorrow unless there are any objections.

**Review by pravindra — APPROVED** [review]

**Comment by kszucs** [comment]
@pravindra I'm fine with this change, mostly because it only affects the gandiva build. In the future we should factor out this change to the level of the submit command.

**Comment by pravindra** [comment]
thanks @kszucs

**Pull request closed without merging**
